dlls/multiplay_gamerules.cpp

```diff
@@ -1316,6 +1318,72 @@ int CountPlayers( void )
 	return num;
 }
 
+//=========================================================
+// Returns whatever survival mode is enabled
+//=========================================================
+BOOL CHalfLifeMultiplay::SurvivalModeEnabled( void )
+{
+	return ( (int)CVAR_GET_FLOAT( "mp_survival_supported" ) == 1 && (int)CVAR_GET_FLOAT( "mp_survival_starton" ) == 1 );
```

Comment:
faster to check the cvar pointers directly than call `CVAR_GET_FLOAT`. I think it's worth doing because SurvivalThink is called every frame.